src/frontend/sharedComponents/EditScreen/ActionButtons.tsx

```diff
@@ -0,0 +1,110 @@
+import React, {useCallback} from 'react';
+import {defineMessages, useIntl} from 'react-intl';
+import {ActionTab} from '../ActionTab';
+import PhotoIcon from '../../images/observationEdit/Photo.svg';
+import AudioIcon from '../../images/observationEdit/Audio.svg';
+import DetailsIcon from '../../images/observationEdit/Details.svg';
+import {BottomSheetModal, useBottomSheetModal} from '../BottomSheetModal';
+import {PermissionAudio} from '../PermissionAudio';
+import {Audio} from 'expo-av';
+import {useNavigationFromRoot} from '../../hooks/useNavigationWithTypes';
+
+const m = defineMessages({
+  audioButton: {
+    id: 'screens.ObservationEdit.ObservationEditView.audioButton',
+    defaultMessage: 'Audio',
+    description: 'Button label for adding audio',
+  },
+  photoButton: {
+    id: 'screens.ObservationEdit.ObservationEditView.photoButton',
+    defaultMessage: 'Photo',
+    description: 'Button label for adding photo',
+  },
+  detailsButton: {
+    id: 'screens.ObservationEdit.ObservationEditView.detailsButton',
+    defaultMessage: 'Details',
+    description: 'Button label to add details',
+  },
+});
+
+interface ActionButtonsProps {
+  showAudio: boolean;
+  fieldIds?: string[];
+}
+
+export const ActionButtons = ({showAudio, fieldIds}: ActionButtonsProps) => {
+  const {formatMessage: t} = useIntl();
+  const navigation = useNavigationFromRoot();
+  const shouldShowAudio = process.env.EXPO_PUBLIC_FEATURE_AUDIO && showAudio;
+  const {
+    openSheet: openAudioPermissionSheet,
+    sheetRef: audioPermissionSheetRef,
+    isOpen: isAudioPermissionSheetOpen,
+    closeSheet: closeAudioPermissionSheet,
+  } = useBottomSheetModal({
+    openOnMount: false,
+  });
+  const [permissionResponse] = Audio.usePermissions({request: false});
+
+  const handleCameraPress = () => {
+    navigation.navigate('AddPhoto');
+  };
+
+  const handleDetailsPress = () => {
+    navigation.navigate('ObservationFields', {question: 1});
+  };
+
+  const handleAudioPress = useCallback(() => {
+    if (!shouldShowAudio) return;
+    if (permissionResponse?.granted) {
+      navigation.navigate('Home', {screen: 'Map'});
+    } else {
+      openAudioPermissionSheet();
+    }
+  }, [
+    navigation,
+    openAudioPermissionSheet,
+    permissionResponse?.granted,
+    shouldShowAudio,
+  ]);
```

Comment:
pressing this causes the app to crash for me

```
Error: Couldn't find a navigation object. Is your component inside NavigationContainer?
```

Reply:
i also end up in this weird state when clearing/recovering from the error. may be related to the issues being addressed via #415

Reply:
https://github.com/digidem/comapeo-mobile/pull/416/commits/9e33768e294d1bc9e5fa9cdd2bddc6ed3e433dac